build/common.rs

```diff
@@ -53,6 +53,7 @@ const ALL_COMPONENTS: &[&str] = &[
     "comp_spiffs_enabled",
     "comp_vfs_enabled",
     "comp_esp_wifi_provisioning_enabled",
+    "comp_esp_ws_client_enabled",
```

Comment:
1. The component is called `esp_websocket_client`, not `esp_ws_client` (the name of the component is the directory name in ESP-IDF). So this one should be `comp_esp_websocket_client_enabled`.
2. This component is no longer available in ESP-IDF V5 (master). It seems since ESP-IDF V5 it is an external component. Therefore, you should include it in this array only if ESP-IDF major version is = 4.

Reply:
> 2. [...] only if ESP-IDF major version is = 4.
Do you have a standard way of specifying this? I can't seem to find one in the codebase.

Reply:
I'm about to release a new version of `esp-idf-sys` where checking the version is no longer necessary, because the available components are no longer hard-coded in `esp-idf-sys`, but scanned from the ESP-IDF directory.